src/apdu_sign.c

```diff
@@ -642,16 +652,15 @@ void finalize_raw_transaction(void) {
                 G.maybe_transaction.v.input_count.snd = G.distinct_input_sources;
                 // Display the complete input amount we are signing, without deducting change
                 G.maybe_transaction.v.amount.fst = G.input_amount.fst;
-                // In plain_output_amount, the change has been deducted
-                G.maybe_transaction.v.amount.snd = G.u.tx.plain_output_amount;
             } else {
                 G.maybe_transaction.v.tag = G.u.tx.is_self_transfer ?
                     OPERATION_TAG_SELF_TRANSFER : OPERATION_TAG_PLAIN_TRANSFER;
-                G.maybe_transaction.v.amount.snd = G.u.tx.plain_output_amount;
             }
-            break;
+            // intentional fall-through
```

Comment:
I think it's better to keep these two separate in case self transfer ever needs special handling.

Reply:
OK